api/app/src/test/kotlin/packit/unit/controllers/PacketControllerTest.kt

```diff
@@ -76,22 +78,30 @@ class PacketControllerTest
         )
     )
 
-    private val htmlContentByteArray = "<html><body><h1>Test html file</h1></body></html>".toByteArray()
-
-    private val inputStream = ByteArrayResource(htmlContentByteArray) to HttpHeaders.EMPTY
-
     private val mockPageablePackets = PageImpl(packets)
-
+    private val testHeaders = HttpHeaders().apply { contentLength = 100L }
     private val packetService = mock<PacketService> {
         on { getPackets(PageablePayload(0, 10), "", "") } doReturn mockPageablePackets
-        on { getFileByHash(anyString(), anyBoolean(), anyString()) } doReturn inputStream
+        on {
+            getFileByHash(
+            anyString(),
+            any<OutputStream>(),
+            any<(HttpHeaders) -> Unit>()
+        )
+        } doAnswer { invocationOnMock ->
+            val outputStream = invocationOnMock.getArgument<OutputStream>(1)
+            outputStream.write("mocked output content".toByteArray())
+            val callback = invocationOnMock.getArgument<(HttpHeaders) -> Unit>(2)
+            callback(testHeaders)
```

Comment:
```suggestion
            val callback = invocationOnMock.getArgument<(HttpHeaders) -> Unit>(2)
            callback(testHeaders)
            val outputStream = invocationOnMock.getArgument<OutputStream>(1)
            outputStream.write("mocked output content".toByteArray())
```

I don't know how much it matters in this unit test, but I think it makes sense to match the order these things happen in the real implementation. In general you can't write HTTP headers (including sending the content length) once the response has been ["committed"](https://docs.oracle.com/javaee/7/api/javax/servlet/ServletResponse.html#isCommitted--)